framework/src/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenesHelper.java

```diff
@@ -383,11 +384,11 @@ private boolean shouldIgnore(
         ImplicitFor implicitFor = elt.getAnnotation(ImplicitFor.class);
         if (implicitFor != null) {
             TypeKind[] types = implicitFor.types();
+            TypeKind atmKind = atm.getUnderlyingType().getKind();
             for (TypeKind tk : types) {
-                if (tk == atm.getKind()) {
-                    return true;
-                }
+                if (tk == atmKind) return true;
             }
+            // TODO: Handle implicitFor.literals()
```

Comment:
Yes, you are right. Whole program inference never annotates literals, only variables. I confused the whole program concepts a bit.